influx-reports/influx.py

```diff
@@ -0,0 +1,140 @@
+             "job_reference='{}' group by time({}s) fill(-1)"
+            ).format(table,job_reference,resolution)
+    all_data = client.query(query)
+    for time_slice in all_data.get_points():
+       for element in time_slice:
+           if element != 'time' and time_slice[element] == 1:
+               uptimes[element.replace('max_','')] += (1 * resolution)
+    return uptimes
+
+def main(args=None):
+    try:
+      influx_ip = os.environ['INFLUX_IP']
+    except KeyError:
+      logging.error("Please set INFLUX_IP")
+      sys.exit(1)
+
+    try:
+      influx_port = os.environ['INFLUX_PORT']
+    except KeyError:
+      logging.error("Please set INFLUX_PORT")
+      sys.exit(2)
+
+    try:
+      job_reference = os.environ['BUILD_TAG']
+    except KeyError:
+      logging.error("Please set BUILD_TAG for its usage as job ref")
+      sys.exit(3)
+
+    client = InfluxDBClient(influx_ip, influx_port, database='telegraf')
+
+    # First find the first and last timestamp from telegraf. This way querying maas_* data will always
+    # be accurate (if no data is reported by maas plugins, this should be a failure metric)
+    find_first_timestamp_query = ("select first(total) from processes "
+                                  "where job_reference = '{}';".format(job_reference))
+    first_ts = return_time(client, find_first_timestamp_query, delta_seconds=5)
+    find_last_timestamp_query = ("select last(total) from processes "
+                                 "where job_reference = '{}';".format(job_reference))
+    last_ts = return_time(client, find_last_timestamp_query, delta_seconds=-5)
+
+    logging.info("Metrics were gathered between {} and {}".format(first_ts, last_ts))
+
+    measurements = ['maas_glance','maas_cinder','maas_keystone', 'maas_heat', 'maas_neutron',
+                    'maas_nova', 'maas_horizon']
+
+    yml_report = False
+    subunit_report = False
+
+    if args.ymlreport:
+      yml_report = True
+      output_file = open(args.ymlreport,'w+')
+      output_file.write("---\n")
+    if args.subunitreport:
+      subunit_report = True
+      output_stream = open(args.subunitreport, 'wb')
+      output = StreamResultToBytes(output_stream)
+      output.startTestRun()
+
+    for measurement in measurements:
+        measurement_data = return_uptime(client, measurement, job_reference)
+        yml_output = yaml.dump({measurement: dict(measurement_data)}, default_flow_style=False)
+        logging.info(yml_output)
+        if yml_report:
+            output_file.write(yml_output)
+        if subunit_report:
+            for measure in measurement_data:
+                # Input status
+                output.status(
+                    test_id=measure,
+                    timestamp=first_ts
+                )
+
+                # Output the end of the event
+                output.status(
+                    test_id=measure,
+                    # TODO(evrardjp): Define a Threshold to compare with
```

Comment:
It isn't useful for now. For this to be useful, we need to:
- add code to find, in a known database of thresholds, the ``measure`` metric threshold.
- Compare this result to the ``measurement_data[measure]`` and output fail/success.

We don't have the code, nor the thresholds yet. That's why TODO.